### PR TITLE
boring 0.6.0 (new formula)

### PR DESCRIPTION
**Pull request by alebeck**
This commit adds the new formula for the `boring` SSH tunnel manager. Boring is a command line tool for managing, starting and stopping SSH tunnels, and keeping them alive in the background.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/b/boring.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  desc "`boring` SSH tunnel manager"
+  homepage "https://github.com/alebeck/boring"
+  url "https://github.com/alebeck/boring/archive/refs/tags/0.6.0.tar.gz"
+  sha256 "460b007759ab2956da2b0f56478cff1574dc668791842c55aa9c2e2c25b03a3d"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args(ldflags: "-X main.version=#{version}"), "./cmd/boring"
```
Comment (chenrui333) [inline Formula/b/boring.rb:11]:
```suggestion
    ldflags = "-s -w -X main.version=#{version}"
    system "go", "build", *std_go_args(ldflags:), "./cmd/boring"
```

File: Formula/b/boring.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  test do
+    output = shell_output(bin/"boring", 1)
+    assert output.start_with?("boring " + version)
```
Comment (chenrui333) [inline Formula/b/boring.rb:16]: maybe some config check test, we need some more substantial test

**Review by chenrui333 — APPROVED** [review]

File: Formula/b/boring.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    ldflags = "-s -w -X main.version=#{version}"
+    system "go", "build", *std_go_args(ldflags:), "./cmd/boring"
+  end
+
+  test do
+    assert_match version.to_s, shell_output(bin/"boring", 1)
+
+    # first interaction with boring should create the user config file
+    # and correctly output that no tunnels are currently configured
+    output = shell_output(bin/"boring list")
```
Comment (carlocab) [inline Formula/b/boring.rb:20]:
```suggestion
    output = shell_output("#{bin}/boring list")
```
`bin/"boring"` is a `Pathname`. `bin/"boring list"` is not.

File: Formula/b/boring.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    # now add an example tunnel and check that it is parsed correctly
+    test_config = testpath/".boring.toml"
+    test_config.write <<~EOF, mode: "a+"
+      [[tunnels]]
+      name = "dev"
+      local = "9000"
+      remote = "localhost:9000"
+      host = "dev-server"
+    EOF
+    output = shell_output(bin/"boring list")
```
Comment (carlocab) [inline Formula/b/boring.rb:33]:
```suggestion
    output = shell_output("#{bin}/boring list")
```

File: Formula/b/boring.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    assert output.strip.end_with?("dev   9000   ->  localhost:9000  dev-server")
```
Comment (carlocab) [inline Formula/b/boring.rb:34]:
```suggestion
    assert_match "dev   9000   ->  localhost:9000  dev-server", output
```

File: Formula/b/boring.rb (hunk 4 above)
Comment (alebeck) [reply Formula/b/boring.rb:33]: updated, thanks.

**Comment by chenrui333** [comment]
@alebeck can you squash the commits into one? Thanks!

**Comment by chenrui333** [comment]
Thanks!

**Review by chenrui333 — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

**Comment by chenrui333** [comment]
@alebeck, thanks for your first contribution to Homebrew-core!! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**